plugins/in_http/http_prot.c

```diff
@@ -510,21 +519,39 @@ static int process_payload(struct flb_http *ctx, struct http_conn *conn,
         type = HTTP_CONTENT_URLENCODED;
     }
 
+    gzip_compressed = flb_is_http_session_gzip_compressed(session);
+    
+    if (gzip_compressed == FLB_TRUE) {
+        gzip_ret = flb_gzip_uncompress(request->data.data, request->data.len, (void **)&uncompressed_data, &uncompressed_size);
+        if (gzip_ret == -1) {
+            flb_error("[http] gzip decompression failed");
+
+            return -1;
+        }
+    } else {
+    uncompressed_data = request->data.data;
+    uncompressed_size = request->data.len;
```

Comment:
One thing, I'd overlooked the indentation glitches. Could you resolve them, then?

Reply:
Fixed.